test/loops.hs

```diff
@@ -1,78 +0,0 @@
-#if !(MIN_VERSION_base(4,11,0))
-import Data.Semigroup ((<>))
-#endif
-import System.IO (stdout, hSetBuffering, BufferMode(LineBuffering))
-import Streamly.Prelude
-
-main :: IO ()
-main = do
-    hSetBuffering stdout LineBuffering
-
-    putStrLn "\nloopTail:\n"
-    drain $ do
-        x <- loopTail 0
-        yieldM $ print (x :: Int)
-
-    putStrLn "\nloopHead:\n"
-    drain $ do
-        x <- loopHead 0
-        yieldM $ print (x :: Int)
-
-    putStrLn "\nloopTailA:\n"
-    drain $ do
-        x <- loopTailA 0
-        yieldM $ print (x :: Int)
-
-    putStrLn "\nloopHeadA:\n"
-    drain $ do
-        x <- loopHeadA 0
-        yieldM $ print (x :: Int)
-
-    putStrLn "\nwSerial:\n"
-    drain $ do
-        x <- (return 0 <> return 1) `wSerial` (return 100 <> return 101)
-        yieldM $ print (x :: Int)
-
-    putStrLn "\nParallel interleave:\n"
-    drain $ do
-        x <- (return 0 <> return 1) `wAsync` (return 100 <> return 101)
-        yieldM $ print (x :: Int)
-
-    where
-
--------------------------------------------------------------------------------
--- Serial (single-threaded) stream generator loops
--------------------------------------------------------------------------------
-
-    -- Generates a value and then loops. Can be used to generate an infinite
-    -- stream. Interleaves the generator and the consumer.
-    loopTail :: Int -> Serial Int
-    loopTail x = do
-        yieldM $ putStrLn "LoopTail..."
-        return x <> (if x < 3 then loopTail (x + 1) else nil)
-
-    -- Loops and then generates a value. The consumer can run only after the
-    -- loop has finished.  An infinite generator will not let the consumer run
-    -- at all.
-    loopHead :: Int -> Serial Int
-    loopHead x = do
-        yieldM $ putStrLn "LoopHead..."
-        (if x < 3 then loopHead (x + 1) else nil) <> return x
```

Comment:
Is this case covered somewhere in property tests? This would be a left fold using the `<>` operation i.e. left associative append. This is especially troublesome in concurrent streams, so we should have tests to make sure it does not get broken.

Reply:
It is covered by the loops test defined in Prelude.Common.